src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -76,15 +73,181 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
+      liveState.removeTask(status.getTaskId());
+      persistentState.removeTaskId(status.getTaskId().getValue());
+      correctCurrentPhase();
+    } else if (isRunningState(status)) {
+      liveState.updateTaskForStatus(status);
+
+      log.info(String.format("Current Acquisition Phase: %s", liveState
+          .getCurrentAcquisitionPhase().toString()));
+
+      switch (liveState.getCurrentAcquisitionPhase()) {
+        case RECONCILING_TASKS :
+          if (liveState.reconciliationComplete()) {
+            reconcilePersistentState();
+            correctCurrentPhase();
+          }
+          break;
+        case JOURNAL_NODES :
+          if (liveState.getJournalNodeSize() == conf.getJournalNodeCount()) {
+            liveState.transitionTo(AcquisitionPhase.NAME_NODE_1);
+          }
+          break;
+        case NAME_NODE_1 :
+          if (liveState.getNameNodeSize() == 1 && liveState.getFirstNameNodeTaskId() != null) {
+            liveState.transitionTo(AcquisitionPhase.NAME_NODE_2);
+          } else {
+            log.info("Cannot locate first namenode task id");
+          }
+          break;
+        case NAME_NODE_2 :
+          if (liveState.getNameNodeSize() == HDFSConstants.TOTAL_NAME_NODES
+              && liveState.getSecondNameNodeTaskId() != null) {
+            reloadConfigsOnAllRunningTasks(driver);
+            //TODO(nicgrayson) need to check if we should format the NN
+            sendMessageTo(
+                driver,
+                liveState.getFirstNameNodeTaskId(), liveState.getFirstNameNodeSlaveId(),
+                HDFSConstants.NAME_NODE_INIT_MESSAGE);
+            blockUntilNameNode1Healthy();
+            sendMessageTo(
+                driver,
+                liveState.getSecondNameNodeTaskId(), liveState.getSecondNameNodeSlaveId(),
+                HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE);
+            liveState.transitionTo(AcquisitionPhase.DATA_NODES);
+          } else {
+            log.info("Cannot locate second namenode task id");
+          }
+          break;
+        case DATA_NODES :
+          break;
+      }
+    } else {
+      log.warn(String.format("Don't know how to handle state=%s for taskId=%s",
+          status.getState(), status.getTaskId().getValue()));
+    }
   }
 
+  @Override
+  public void resourceOffers(SchedulerDriver driver, List<Offer> offers) {
+    log.info(String.format("Received %d offers", offers.size()));
+
+    if (liveState.getStagingTasksSize() != 0) {
+      log.info("Declining offers because tasks are currently staging");
+      for (Offer offer : offers) {
+        driver.declineOffer(offer.getId());
+      }
+    } else {
+      boolean acceptedOffer = false;
+      for (Offer offer : offers) {
+        if (acceptedOffer) {
+          driver.declineOffer(offer.getId());
+        } else {
+          switch (liveState.getCurrentAcquisitionPhase()) {
+            case RECONCILING_TASKS :
+              if (liveState.reconciliationComplete()) {
```

Comment:
Could we add the state machine to a separate class to be more object oriented and so the scheduler doesn't grow to be too large?